File: frontend/app/models/coursePlanItem.ts
import type { BaseTimestamp } from "./baseTimestamp";
import { CoursePlanItemStatusEnum } from "../constants/coursePlanItemStatusEnum";
import type { Course } from "./course";

export interface CoursePlanItem extends BaseTimestamp {
    coursePlanId: number;
    courseId: number;
    status: CoursePlanItemStatusEnum;
    remarks?: string | null;
    course?: Course;
}

File: endpoint/models/course_plan_item.go
package models

import (
	"yosimaril/CourseEnrollment/constants"
)

type CoursePlanItem struct {
	CoursePlanID uint                           `json:"course_plan_id" gorm:"primaryKey"`
	CourseID     uint                           `json:"course_id" gorm:"primaryKey"`
	Status       constants.CoursePlanItemStatus `json:"status"`
	Remarks      *string                        `json:"remarks"`

	BaseModel

	CoursePlan CoursePlan `gorm:"foreignKey:CoursePlanID"`
	Course     Course     `gorm:"foreignKey:CourseID"`
}
